include/solvers/time_solver.h

```diff
@@ -28,6 +28,10 @@
 #include "libmesh/solution_history.h"
 #include "libmesh/qoi_set.h"
 
+// Sensitivity Calculation related includes
+#include "libmesh/parameter_vector.h"
+#include "libmesh/sensitivity_data.h"
```

Comment:
Let's use forward declarations for these so we don't bring in more header dependencies?

Reply:
The sensitivity data include is needed, but removed the parameter vector header.

Reply:
No, neither are needed, and both need forward declarations.  I'll fix it.

Reply:
Wow... and while I'm at it I'll fix the pre-existing problem with a couple other headers here.  Even the things we have as unique_ptr members can be forward declared in cases like this where we don't have any constructors or destructors in the header or implicit.

Reply:
I tried the forward declaration for the SensitivityData, but got compilation errors on my local machine.

Reply:
Yeah; typically when you put a forward declaration into a header instead of a full include, you'll find that source files (and sometimes other headers) which actually needed that full include were getting it indirectly when they should have been including it themselves.  So you put in the inclusions where they're needed while taking them away from where they're not.  I think I've got that fixed with this last commit.